python/ml4ir/base/features/feature_fns/base.py

```diff
@@ -0,0 +1,46 @@
+from tensorflow.keras import layers
+
+from ml4ir.base.io.file_io import FileIO
+
+DEFAULT_VALUE = "default_value"
+
+
+class BaseFeatureLayerOp(layers.Layer):
+    """Abstract feature layer operation class"""
```

Comment:
One way to enforce `LAYER_NAME` in all subclasses is to have something like:
```python
    @property
    @abstractmethod
    def LAYER_NAME(self):
        raise AttributeError
```

in this class